src/realm/object-store/c_api/util.hpp

```diff
@@ -115,6 +116,33 @@ struct FreeUserdata {
 };
 
 using UserdataPtr = std::unique_ptr<void, FreeUserdata>;
+
+template <typename... Args>
+class CallbackRegistry {
```

Comment:
I added a comment to that effect. I won't elaborate on this in the public api because operations on `realm_t*` (and other database objects) are not thread-safe to begin with.